flask-server/main.py

```diff
@@ -74,6 +76,88 @@ def login_user():
         }
         return response_object, 500
 
+# logout api
+@app.route('/apis/logout', methods=['POST'])
+def logout_user():
+    user = request.json['user']
+    if user:
+        auth_token = user.get('Authorization')
+    else:
+        auth_token = ''
+    if auth_token:
+        resp = decode_auth_token(auth_token)
+        if is_valid_instance(resp):
+            return save_token(token=auth_token)
+        else:
+            response_object = {
+                'status': 'fail',
+                'message': resp
+            }
+            return response_object, 401
+    else:
+        response_object = {
+            'status': 'fail',
+            'message': 'Provide a valid auth token.'
+        }
+        return response_object, 403
+
+def is_valid_instance(resp):
+    if resp == "Token denylisted. Please log in again" or resp == "Signature expired. Please log in again." or resp == "Invalid token. Please log in again.":
+        return False
+    return True
+
+# add token to denylist
+def save_token(token):
+    try:
+        deny_token = datastore.Entity(key=datastore_client.key('denylist_token'))
+        deny_token.update({
+            'jwt': str(token)
+        })
+        datastore_client.put(deny_token)
+
+        response_object = {
+            'status': 'success',
+            'message': 'Successfully logged out.'
+        }
+        return response_object, 200
+    except Exception as e:
+        response_object = {
+            'status': 'fail',
+            'message': e
```

Comment:
Directly sending the error back to the client might cause issues such as sending server side errors explicitly to the client. Would prefer hardcode an error message here.